driver_test.go

```diff
@@ -0,0 +1,106 @@
+package sqlair_test
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"strings"
+	"sync"
+	"unsafe"
+
+	"github.com/mattn/go-sqlite3"
+)
+
+// This file contains a wrapper sql.Driver over the SQLite driver which
+// monitors the creation and closing of prepared statements and stores the
+// references to said statements. We can later use that information to check
+// for statement leaks.
+
+// The maps below store the pointers to the created/closed statements indexed
+// by test case. We use unsafe pointers instead of references to the objects
+// because if we stored a reference the runtime.Finalizer would not be able to
+// run.
+var openedStmts = map[string]map[uintptr]string{}
```

Comment:
It would be nice to have these as members of the suite itself, but I can't see an easy way to supply them to the driver without repeating register/unregister.